tools/image_database_to_video/image_database_to_video.cc

```diff
@@ -0,0 +1,173 @@
+// BoB robotics includes
+#include "common/macros.h"
+#include "common/path.h"
+#include "common/string.h"
+#include "navigation/image_database.h"
+
+// TBB
+#include "tbb/parallel_for_each.h"
+
+// Third-party includes
+#include "third_party/CLI11.hpp"
+#include "third_party/path.h"
+
+// Standard C++ includes
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <string>
+
+using namespace BoBRobotics;
+using namespace Navigation;
+
+// **TODO**: Allow user to choose output video format
+constexpr const char *VideoExtension = ".mp4";
```

Comment:
We are archiving the originals (I've got a copy and so has @tenxlenx) but we just want a convenient format for uploading to GitHub or passing around on a USB drive or whatever (having millions of little image files makes things reaaaaally slow).

The problem with reducing the resolution of these wrapped images is that I suspect it'll reduce the quality of the unwrapped images generated from it. If you do this, then there effectively two places where you're throwing away information: first when you reduce the resolution and then when you unwrap it. Alright, *technically* you're throwing away some information by converting to MP4 but it really shouldn't be much...

Reply:
But MJPEG would still be a single file, it would just use frame-based compression rather than temporal compression like mp4 (I feel we had this discussion before and made that the default for the image database tool)

Reply:
Yeah we did make it the default for the image database tool. The thing is I _want_ the temporal compression here as it makes sense for these sorts of route-type databases (i.e. because the view changes continuously over time without the sorts of big jumps you get in grid-type databases). I don't really want to convert all of our databases into video files. I just want to do it for this particular dataset where we have these really long routes with lots of images in. That way you can have the high-res raw videos on GitHub and then just bundle a script for unwrapping with our tool and it would be at that stage where we throw away data by dropping frames/reducing the resolution etc.

We could make the file format optional to make this tool a bit more general purpose (maybe it _would_ be useful to use MJPEG for some other databases so we don't have so many files...).